compiler/src/iree/compiler/Codegen/Transforms/Transforms.cpp

```diff
@@ -506,17 +506,30 @@ void moveLoopInvariantCodeFromGuaranteedLoops(Operation *target) {
         loopLike.getLoopLowerBounds();
     std::optional<SmallVector<OpFoldResult>> maybeUpperBounds =
         loopLike.getLoopUpperBounds();
-    if (!maybeLowerBounds || !maybeUpperBounds) {
+    std::optional<SmallVector<Value>> maybeIvs =
+        loopLike.getLoopInductionVars();
+    if (!maybeLowerBounds || !maybeUpperBounds || !maybeIvs) {
       return;
     }
 
     // If any lower + upper bound pair cannot be definitely verified as lb < ub
     // then the loop may have a zero trip count.
-    for (auto [lb, ub] :
-         llvm::zip_equal(*maybeLowerBounds, *maybeUpperBounds)) {
-      if (!ValueBoundsConstraintSet::compare(lb, ValueBoundsConstraintSet::LT,
-                                             ub)) {
-        return;
+    for (auto [lb, ub, iv] :
+         llvm::zip_equal(*maybeLowerBounds, *maybeUpperBounds, *maybeIvs)) {
+      if (iv.getType().isIndex()) {
```

Comment:
Since you're comparing the `lb` and `ub`, you can just verify that those types are index instead of getting + iterating the ivs.

Reply:
`lb` and `ub` are both `OpFoldResult`s - there's no easy way to get the type, especially since, if they're both constants, we might want to "upgrade" them to index anyway. But do let me know if I'm missing something.

Reply:
Ah that's a good point, I didn't think about that. This looks good then.